Bottom: ad1819f79ea1bc285bc5de8280974077e1d1c19c
Top:    61326ea1fb69c26d985d4d865c884af400c08e56
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-20 14:02:42 +0000

More hack to avoid GL calls when we're out of context


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 748af2c..5ce5ac1 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -505,6 +505,12 @@ ghid_set_color (hidGC gc, const char *name)
 
   gc->colorname = (char *) name;
 
+  if (!check_gl_drawing_ok_hack)
+    {
+      current_color = NULL;
+      return;
+    }
+
   if (gport->colormap == NULL)
     gport->colormap = gtk_widget_get_colormap (gport->top_window);
   if (strcmp (name, "erase") == 0)
